### ISSUE TITLE
haml files + RuboCop

### ISSUE THREAD

**Issue by jufemaiz** [AtomLinter/linter-rubocop#145]
Currently seeing RuboCop errors when in HAML files.

<img width="910" alt="screen shot 2016-07-21 at 4 28 05 pm" src="https://cloud.githubusercontent.com/assets/16481/17013216/569d879c-4f60-11e6-8a33-1ea600a16e95.png">

```
Scopes at cursor:
- text.haml
- meta.tag.haml
- entity.name.tag.haml
```

RuboCop error:

```
%ul: unknown type of percent-literal
```

**Comment by Arcanemagus** [comment]
Well now I'm just confused... none of those scopes are in [the list](https://github.com/AtomLinter/linter-rubocop/blob/v0.4.7/index.coffee#L42-L47) of scopes this package should trigger on...

**Comment by Arcanemagus** [comment]
For reference, this was originally filed and discussed over in https://github.com/AtomLinter/linter-haml/issues/99.

**Comment by jufemaiz** [comment]
@Arcanemagus me too ><

It's only periodic, so I'm really confused to the trigger. That I haven't determined yet. Once it happens, I need to close the project and reopen.

**Comment by quolpr** [comment]
+1, having this problem too. After file closing all become fine. I am not sure, but it appeared after last update.

**Comment by Arcanemagus** [comment]
The only way I can think of for this to happen is if _part_ of the file is marked as one of the [allowed scopes](https://github.com/AtomLinter/linter-rubocop/blob/v0.4.7/index.coffee#L42-L47).

Do you have a file you consistently see this on that you can share? What do you get when you run `Editor: Log Cursor Scopes` _immediately_ after you start seeing `RuboCop` messages in your Haml file?

**Comment by 12eh** [comment]
I can reliably reproduce this by doing string interpolation in an attribute hash. Create a new haml file:
```
%a{ href: "asdfasdf" }
```
All good so far. Now,
```
%a{ href: "asdfasdf#{}" }
```
...and the rubocop linter kicks in. Save the file, close and reopen, rubocop errors are gone and I can continue editing. Until I do another string interpolation, anyway.

**Comment by Arcanemagus** [comment]
Ah ha!

In this block:

```haml
%a{ href: "asdfasdf#{}" }
```

If you place the cursor in between the `{}` in the string interpolation part, one of the scopes is `source.ruby`!

Can somebody who knows HAML better file an issue on [language-haml](https://github.com/ezekg/language-haml) to get that marked as `source.embedded.ruby` (or just remove `source.ruby` as it already has `meta.embedded.line.ruby`)?

**Comment by maschwenk** [comment]
Looks like this was resolved upstream

**Comment by joelmichael** [comment]
Seems to still be happening to me.

**Comment by vzamanillo** [comment]
> Seems to still be happening to me.

#48
